apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -197,7 +224,48 @@ NetSimRouterLogModal.prototype.isVisible = function () {
  * Fill the root div with new elements reflecting the current state
  */
 NetSimRouterLogModal.prototype.render = function () {
-  // Be lazy, don't render if not visible.
+  if (usingNewLogBrowser()) {
+    this.newRender_();
+  } else {
+    this.oldRender_();
+  }
+};
+
+NetSimRouterLogModal.prototype.newRender_ = function () {
+  const getOriginNodeName = entry => {
+    const originNode = entry.getOriginNode();
+    return originNode ? originNode.getDisplayName() : entry.nodeID;
+  };
+  const tableRows = this.getSortedFilteredLogEntries(this.logEntries_).map(entry => ({
+    'uuid': entry.uuid,
+    'timestamp': entry.timestamp,
+    'logged-by': getOriginNodeName(entry),
+    'status': entry.getLocalizedStatus(),
+    'from-address': entry.getHeaderField(Packet.HeaderType.FROM_ADDRESS),
+    'to-address': entry.getHeaderField(Packet.HeaderType.TO_ADDRESS),
+    'packet-info': entry.getLocalizedPacketInfo(),
+    'message': entry.getMessageAscii()
+  }));
+  ReactDOM.render(
+    <NetSimLogBrowser
+      isOpen={this.isVisible()}
+      handleClose={this.hide.bind(this)}
+      i18n={i18n}
```

Comment:
would it not be easier to have the react components just require this themselves, rather than passing it down through the entire requirements chain?